echo/README.md

```diff
@@ -1,5 +1,224 @@
+
+    ```bash
+    $ wasmcloud -m manifest.yaml
+    ```
+
+## Using `wash` for better developer experience
+
+In this second part, we will spin a NATS server and a local Docker registry to spice things up.
+
+1. We will run NATS servers and Docker's local image registry as our prerequisites.
+
+    NATS is a message broker used by wasmcloud’s self-managing lattice network. The image registry is for to store our WebAssembly modules.
+
+    ```yaml
+    version: "3"
+    services:
+      registry:
+        image: registry:2
+        ports:
+        - "5000:5000"
+      nats:
+        image: nats:2.1.9
+        ports:
+          - "6222:6222"
+          - "4222:4222"
+          - "8222:8222"
+    ```
+
+    You can find `docker-compose.yml` in the project’s directory. We will run our containers in detached mode to reduce clutter.
+
+    ```bash
+    $ docker-compose up -d
+    ```
+
+    Start a repl session using `wash up` command:
+
+    ```bash
+    $ wash up
+    ```
+
+    Check if session is running:
+
+    ```bash
+    $ ctl get hosts
+    ```
+
+    It will output something like:
+
+    ```bash
+    Host ID                                                   Uptime (seconds)
+    NDGPVEOZKAYT6JRL2C4BDVSW3PBBOYTWRLC2QFCSLXWDX4KCIRHM6WCZ  1357
+    ```
+
+2. Build .wasm file and sign it using make file:
+
+    ```bash
+    $ cargo build --release
+    $ wash claims sign ./target/wasm32-unknown-unknown/release/echo.wasm -c wasmcloud:httpserver --name "echo" --ver 0.2.2
+    ```
+
+    Or you can use project’s make file:
+
+    ```bash
+    make release VERSION=0.2.2
+    ```
+
+    This will produce `echo.wasm` actor and its signed version `echo_s.wasm`.
+
+3. Push signed actor into to local registry using `wash`:
+
+    ```bash
+    wash reg push localhost:5000/echo:0.2.2 ./target/wasm32-unknown-unknown/release/echo_s.wasm --insecure
+    ```
+
+4. Run it:
+
+    ```bash
+    ctl start actor localhost:5000/echo:0.2.2
+    ```
+
+    Steps from now on is independent of our current example but we keep it for completeness:
+
+5. Start your capability that runs an html server:
+
+    ```bash
+    ctl start provider wasmcloud.azurecr.io/httpserver:0.11.1
+    ```
+
+    Check out https://wasmcloud.dev/app-dev/create-provider for how to write your own:
+
+6. Link actor with capability using `wash`:
+
+    ```bash
+    // Get actor and capability provider ids using:
+    // ctl get inventory «HOST_ID»
+
+    $ ctl get inventory NDGPVEOZKAYT6JRL2C4BDVSW3PBBOYTWRLC2QFCSLXWDX4KCIRHM6WCZ
+
+    // Establish a contract between them using link command
+    // Pass along environment variables required by the capability provider using:
+    // ctl link «ACTOR_ID» «PROVIDER_ID» «CONTRACT_ID» PORT=8080
+
+    $ ctl link MCUDTMOOZCVAM5EBNN4U3X2OGHNIY3BKPEW66HY4RTCYYVWXOE7ESVDQ VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4A YJ5RKAX2UNSCAPHA5M wasmcloud:httpserver PORT=8080 
+    ```
+
+7. Visit `http://localhost:8080` or send a request:
```

Comment:
Note here that the `curl` will need to be in a separate terminal, while the `call` will have to be in the REPL terminal window